src/Libraries/Mailer/Adapters/SendgridAdapter.php

```diff
@@ -16,30 +16,131 @@
 
 use Quantum\Libraries\Mailer\MailerInterface;
 use Quantum\Libraries\Curl\HttpClient;
+use Quantum\Libraries\Mailer\MailTrap;
+use Exception;
 
+/**
+ * class SendgridAdapter
+ * @package Quantum\Libraries\Mailer
+ */
 class SendgridAdapter implements MailerInterface
 {
+    use MailerAdapterTrait;
+
+    /**
+     * @var string
+     */
+    public $name = 'Sendgrid';
+
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @var string
+     */
+    private $apiUrl = 'https://api.sendgrid.com/v3/mail/send';
+
+    /**
+     * @var array
+     */
+    private $data = [];
+
+    /**
+     * @var SendgridAdapter|null
+     */
+    private static $instance = null;
+
+    /**
+     * SendgridAdapter constructor
+     * @param array $params
+     */
+    private function __construct(array $params)
+    {
+        $this->httpClient = new HttpClient();
+
+        $this->apiKey = $params['api_key'];
+    }
+
+    /**
+     * Get Instance
+     * @param array $params
+     * @return SendgridAdapter
+     */
+    public static function getInstance(array $params): SendgridAdapter
+    {
+        if (self::$instance === null) {
+            self::$instance = new self($params);
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * Prepares the data
+     */
+    private function prepare()
+    {
+        $this->data['from'] = [
+            'email' => $this->from['email']
+        ];
+
+        $this->data['personalizations'] = [
+            ['to' => $this->addresses]
+        ];
+
+        if ($this->subject) {
+            $this->data['subject'] = $this->subject;
+        }
+
+        if ($this->message) {
+            if ($this->templatePath) {
+                $body = $this->createFromTemplate();
+            } else {
+                $body = is_array($this->message) ? implode($this->message) : $this->message;
+            }
+
+            $this->data['content'] = [
+                'type' => 'text/html',
+                'value' => trim(str_replace("\n", "", $body))
+            ];
```

Comment:
data['content '] should be in array

$this->data['content'] = [
                [
                    'type' => 'text/html',
                    'value' => trim(str_replace("\n", "", $body))
                ]
            ];